src/main/java/com/faforever/client/mod/ModService.java

```diff
@@ -110,7 +111,14 @@ public void afterPropertiesSet() {
         onModDirectoryReady();
       }
     };
-    modDirectoryChangedListener.invalidated(preferencesService.getPreferences().getForgedAlliance().modsDirectoryProperty());
+    taskService.submitTask(new CompletableTask<Void>(Priority.LOW) {
+      @Override
+      protected Void call() throws Exception {
+        updateTitle(i18n.get("modVault.loadingMaps"));
+        modDirectoryChangedListener.invalidated(preferencesService.getPreferences().getForgedAlliance().modsDirectoryProperty());
+        return null;
+      }
+    });
```

Comment:
Why use a task to invalidate manually? Why is this different than the maps?

Reply:
Cause it should be done in the background and maps are also loaded with a task